packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -619,21 +622,20 @@ function flush_queued_effects(effects) {
 				}
 			}
 
-			// if state is written in a user effect, abort and re-schedule, lest we run
-			// effects that should be removed as a result of the state change
-			if (
-				current_batch !== null &&
-				current_batch.current.size > n &&
-				(effect.f & USER_EFFECT) !== 0
-			) {
-				break;
+			if (eager_block_effects.length > 0) {
+				// TODO this feels incorrect! it gets the tests passing
+				old_values.clear();
```

Comment:
This is the correct place since it's equivalent timing to before: previously you would bail the loop and start a new flush. That means you ended up at the end of the `flush_effects` loop where we also call `clear()`, and then did another flush.
I have some nagging doubts if the logic in general for this is sound but we can tackle that later.